### PR TITLE
smartos_image_facts: rename to smartos_image_info

### LINKED ISSUE #60532: smartos_image_facts needs to be renamed to _info and stop returning ansible_facts
##### SUMMARY
According to the [requirements](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_checklist.html#contributing-to-ansible-objective-requirements), `_facts` modules should return information that is specific to the host the module runs on as `ansible_facts`. (See also ansible/ansible#54280 for some discussion.)

I think that this module do not satisfy this condition: its return value depends on the module option `filter`. Please verify whether this is actually the case.

To fix this, the core team thinks the best approach is to rename the `_facts` modules to `_info` modules, and change the return value so that if the module is called with the `_facts` name (should be deprecated, and will then be removed in Ansible 2.13), it should return `ansible_facts` as now (and print a deprecation message), and if it is called with the `_info` name, it no longer returns `ansible_facts`, so that the result needs to be registered. This informs users that they have to change something, won't break their playbooks, and allows them (by switching to the `_info` names) to verify that their roles and playbooks no longer use the facts.

If that approach doesn't work, or you want to change the module's interface (like rename the return value(s)), it is better to deprecate the `_facts` module and create a new `_info` module next to it. To see both approaches implemented, you can take a look at https://github.com/ansible/ansible/commit/2d98734ad51c3de557c3a7d1f7166bdd114ceb0f (ansible/ansible#60178).

If you have any input — preferred method to solve this, doubts that all modules should be renamed, ... — please write it here!

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
smartos_image_facts

##### ANSIBLE VERSION
```paste below
2.9.0
```

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #60532.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
smartos_image_facts

### CODE REVIEW

**Comment by anshulbehl** [comment]
Docs Review done-LGTM.

**Review by samccann — COMMENTED** [review]
docs portion lgtm

File: lib/ansible/modules/cloud/smartos/smartos_image_info.py [hunk 1]
```diff
@@ -33,21 +34,28 @@
 
 EXAMPLES = '''
 # Return facts about all installed images.
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/smartos/smartos_image_info.py]:
```suggestion
 # Return information about all installed images.
```

File: lib/ansible/modules/cloud/smartos/smartos_image_info.py [hunk 2]
```diff
@@ -33,21 +34,28 @@
...
-- smartos_image_facts:
+- smartos_image_info:
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/smartos/smartos_image_info.py:37]:
```suggestion
- smartos_image_info:
  register: result
```

File: lib/ansible/modules/cloud/smartos/smartos_image_info.py [hunk 3]
```diff
@@ -33,21 +34,28 @@
...
 
 # Show, how many clones does every image have.
-- smartos_image_facts:
+- smartos_image_info:
+  register: result
 
-- debug: msg="{{ smartos_images[item]['name'] }}-{{smartos_images[item]['version'] }}
+- debug: msg="{{ result.smartos_images[item]['name'] }}-{{ result.smartos_images[item]['version'] }}
+            has {{ result.smartos_images[item]['clones'] }} VM(s)"
+  with_items: "{{ result.smartos_images.keys() }}"
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/smartos/smartos_image_info.py]: Do you mind also updating this to work with python 2 and 3?
```suggestion
  with_items: "{{ result.smartos_images.keys() | list }}"
```

File: lib/ansible/modules/cloud/smartos/smartos_image_info.py [hunk 4]
```diff
@@ -33,21 +34,28 @@
...
+
+# When the module is called as smartos_image_facts, return values are published
+# in ansible_facts['smartos_images'] and can be used as follows.
+# Note that this is deprecated and will stop working in Ansible 2.13.
+- debug: msg="{{ smartos_images[item]['name'] }}-{{ smartos_images[item]['version'] }}
             has {{ smartos_images[item]['clones'] }} VM(s)"
   with_items: "{{ smartos_images.keys() }}"
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/smartos/smartos_image_info.py]:
```suggestion
   with_items: "{{ smartos_images.keys() | list }}"
```

File: lib/ansible/modules/cloud/smartos/smartos_image_info.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/modules/cloud/smartos/smartos_image_info.py]: Thanks! Fixed.

File: lib/ansible/modules/cloud/smartos/smartos_image_info.py (hunk 2 above)
Comment (felixfontein) [reply lib/ansible/modules/cloud/smartos/smartos_image_info.py:37]: Changed. (Also the next one.)

File: lib/ansible/modules/cloud/smartos/smartos_image_info.py (hunk 3 above)
Comment (felixfontein) [reply lib/ansible/modules/cloud/smartos/smartos_image_info.py]: Good idea. Actually, shouldn't it also work without the `.keys()`? It is definitely more explicit with `.keys()`, though.

Comment (s-hertel) [reply lib/ansible/modules/cloud/smartos/smartos_image_info.py]: True, good point.

**Comment by felixfontein** [comment]
@anshulbehl @samccann @s-hertel thanks a lot for reviewing this one!

**Pull request merged**